src/pipecat/services/inworld/tts.py

```diff
@@ -0,0 +1,600 @@
+            # ================================================================================
+            # Use aiohttp to make request to Inworld's endpoint
+            # Behavior differs based on streaming mode
+            async with self._session.post(
+                self._base_url, json=payload, headers=headers
+            ) as response:
+                # ================================================================================
+                # STEP 4: HANDLE HTTP ERRORS
+                # ================================================================================
+                # Check for API errors (expired keys, invalid requests, etc.)
+                if response.status != 200:
+                    error_text = await response.text()
+                    logger.error(f"Inworld API error: {error_text}")
+                    await self.push_error(ErrorFrame(f"Inworld API error: {error_text}"))
+                    return
+
+                # ================================================================================
+                # STEP 5: PROCESS RESPONSE (MODE-SPECIFIC)
+                # ================================================================================
+                # Choose processing method based on streaming mode
+                if self._streaming:
+                    # Stream processing: JSON line-by-line with real-time audio
+                    async for frame in self._process_streaming_response(response):
+                        yield frame
+                else:
+                    # Non-stream processing: Complete JSON response with batch audio
+                    async for frame in self._process_non_streaming_response(response):
+                        yield frame
+
+            # ================================================================================
+            # STEP 6: FINALIZE METRICS AND CLEANUP
+            # ================================================================================
+            # Start usage metrics tracking after successful completion
+            await self.start_tts_usage_metrics(text)
+
+        except Exception as e:
+            # ================================================================================
+            # STEP 7: ERROR HANDLING
+            # ================================================================================
+            # Log any unexpected errors and notify the pipeline
+            logger.error(f"{self} exception: {e}")
+            await self.push_error(ErrorFrame(f"Error generating TTS: {e}"))
+        finally:
+            # ================================================================================
+            # STEP 8: CLEANUP AND COMPLETION
+            # ================================================================================
+            # Always stop metrics tracking, even if errors occurred
+            await self.stop_ttfb_metrics()
```

Comment:
This is where you want to stop all metrics, which calls:
```
    async def stop_all_metrics(self):
        """Stop all active metrics collection."""
        await self.stop_ttfb_metrics()
        await self.stop_processing_metrics()
```

stop_ttfb_metrics was likely called earlier, so it's redundant, but this will also stop the processing metrics.

```suggestion
            await self.stop_all_metrics()
```